### ISSUE TITLE
Read router params in the title property using string interpolation

### ISSUE THREAD

**Issue by MonaGamal** [angular/angular#48204]
### Description

Is it possible to just read the params of the router with string interpolation without making a resolver or injecting the Title?

### Proposed solution

```
{
    path: ':productId',
    component: ProductComponent,
    title: `Product details: ${:productId}`
  }
```

### Alternatives considered

Or any other better solution?

**Comment by atscott** [comment]
No, _but_ this is so trivial with functional resolvers that special handling for template strings is not necessary: 
```
{
    path: ':productId',
    component: ProductComponent,
    title: (route) => `Product details: ${route.params['productId']}`
  }
```
stackblitz example: https://stackblitz.com/edit/angular-ivy-pz6chj?file=src%2Fmain.ts,src%2Findex.html

This was a easy to answer quickly but for the record, this doesn't fall into the bug report or feature request category. This issue tracker is not suitable for support requests. In the future, please post your questions on [StackOverflow](http://stackoverflow.com/) using tag `angular`.

If you are wondering why we don't resolve support issues via the issue tracker, please [check out this explanation](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-got-a-question-or-problem).

**Comment by MonaGamal** [comment]
@atscott  Thank you for your reply and for the stackblitz sample. 

That is exactly what we were looking for. It seems we have tried to make it in Angular 14 and it wasn't supported at that time as far as I understand.

Angular 14
`title?: string | Type<Resolve<string>>;`

Angular 15
`title?: string | Type<Resolve<string>> | ResolveFn<string>;
`

**Comment by atscott** [comment]
@MonaGamal Functional guards and resolvers are available in v14.2. If your application is not ready to upgrade to v15, you can still get the feature in 14.2
